book_source/03_intermediate_users_guide/01_pecan_xml.Rmd

```diff
@@ -1,195 +1,487 @@
+		<scratchdir>/tmp/carya</scratchdir>
+		<clearscratch>TRUE</clearscratch>
+		<qsub>qsub -N @NAME@ -o @STDOUT@ -e @STDERR@ -S /bin/bash</qsub>
+		<qsub.jobid>Your job ([0-9]+) .*</qsub.jobid>
+		<qstat>qstat -j @JOBID@ &> /dev/null || echo DONE</qstat>
+		<job.sh>module load udunits R/R-3.0.0_gnu-4.4.6</job.sh>
+	</host>
 ```
 
-The section meta.analysis needs to exists for a meta.analysis to be executed, even though all tags inside are optional.
+The `host` section has the following tags:
 
-* **iter** : [optional] [MCMC](http:/en.wikipedia.org/wiki/Markov_chain_Monte_Carlo) (Markov Chain Monte Carlo) chain length, i.e. the total number of posterior samples in the meta-analysis, default is 3000. Smaller numbers will run faster but produce larger errors.  
-* **random.effects** : [optional] Whether to include random effects (site, treatment) in meta-analysis model. Can be set to FALSE to work around convergence problems caused by an over parameterized model (e.g. too many sites, not enough data). The default value is TRUE.
-* **update** : [optional] Should previous results of meta.analysis and get.traits be re-used. If set to TRUE the meta-analysis and get.trait.data will always be executed. Setting this to FALSE will try and reuse existing results. Future versions will allow for AUTO as well which will try and reuse if the PFT/traits have not changed. The default value is FALSE.
-* **threshold** threshold for Gelman-Rubin convergence diagnostic (MGPRF); default is 1.2
+* `name`: [optional] name of host server where model is located and executed, if not specified localhost is assumed.
+* `rundir`: [optional/required] location where all the configuration files are written. For localhost this is optional (`<outdir>/run` is the default), for any other host this is required.
+* `outdir`: [optional/required] location where all the outputs of the model are written. For localhost this is optional (`<outdir>/out` is the default), for any other host this is required.
+* `scratchdir`: [optional] location where output is written. If specified the output from the model is written to this folder and copied to the outdir when the model is finished, this could significantly speed up the model execution (by using local or ram disk).
+* `clearscratch`: [optional] if set to TRUE the scratchfolder is cleaned up after copying the results to the outdir, otherwise the folder will be left. The default is to clean up after copying.
+* `qsub`: [optional] the command to submit a job to the queuing system. There are 3 parameters you can use when specifying the qsub command, you can add additional values for your specific setup (for example -l walltime to specify the walltime, etc). You can specify @NAME@ the pretty name, @STDOUT@ where to write stdout and @STDERR@, where to write stderr. You can specify an empty element (`<qsub/>`) in which case it will use the default value is `qsub -V -N @NAME@ -o @STDOUT@ -e @STDERR@ -s /bin/bash`.
```

Comment:
Don't we have more documentation elsewhere about how to set up remote and qsub? Seems like there should be a link to that here as this snippet is not sufficient to allow most people to figure out how to set up and debug/test their qsub configuration.

Reply:
Possibly. I remember writing more extensive `remote` documentation at some point, so I'll try to find that and link it here.

Reply:
Resolved in 04e95e45b